### ISSUE TITLE
Support a multi-valued Accept-header and add quality factors to content negotiation

### ISSUE THREAD

**Issue by ThijsBroersen** [finagle/finch#1059]
As a user I would like Finch to take into account a multi-valued Accept-header. Currently it ignores all values when the Accept-header is multi-valued.

I would also like quality factors within an Accept-header to be respected.
E.g. 
```
Input
  .get("/test")
  .withHeaders("Accept" -> "text/plain;q=0.3,application/json;q=0.9")
```
Currently the quality factors are ignored.

**Comment by vkostyukov** [comment]
@ThijsBroersen Thanks for the ticket!

Not sure I follow the first part about being a multi-value. This should be supported (only qualities are ignored) and if not, it's a bug. Mind providing more details on this?

**Comment by ThijsBroersen** [comment]
@vkostyukov you are correct, it takes the first Content-Type that matches any value from the Accept-header. My tests were off...
